### PR TITLE
PyPI release

### PR DESCRIPTION
**Pull request by enrico-lattuada**
This PR implements the changes needed to release the package on PyPI.

Closes issue #7

### CODE REVIEW

**Comment by enrico-lattuada** [comment]
I tried manually uploading the source distribution to TestPyPI (using versions from ``v0.0.0`` to ``v0.0.4``).
I tested it and it works on different platforms and environments.

This is no different than downloading and installing the package from the repository, but it allows us to run tests before the final release to PyPI.

I added a workflow that should do the same.
Currently, it runs only when a new release is published.

In the next future, we should:
* Decide when to run the workflow
* Yank the previous test releases

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (krauthex) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: I have no particular issue here, I'm just wondering why all other templates are `.yml` and this one is `.md`? any particular reason for this? :-)

File: docs/source/pypi.rst
Comment (krauthex) [inline docs/source/pypi.rst:1]: shouldn't this contain a guide on how it works? you mentioned in the PR that it does work from the TestPyPi so can we add the installation guide here as well?

**Review by krauthex — COMMENTED** [review]
Looks good to me! :-) I'd appreciate a short feedback on my two comments, but I don't see any other issue here at the moment. thanks for testing all this! :-) really feels like this is coming together now, well done!

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (enrico-lattuada) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: As far as I understand, it just prefills the (markdown) message when a new pull request is opened (see [here](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository)).

The `.yml` files create instead a more structured page with text areas etc.

File: docs/source/pypi.rst
Comment (enrico-lattuada) [reply docs/source/pypi.rst:1]: Sure, I was planning to explain the installation process in the doc page once the PyPI package is released, but it's good to have at least here a short list of the operations to perform to install from TestPyPI for future reference.
I will write it in the next comment.

**Comment by enrico-lattuada** [comment]
Currently, we have a source distribution available on TestPyPI.
The wheels will be addressed in the next release.
The installation process works as described in the ``Build from source`` guide, except that the package is downloaded from TestPyPI and not from GitHub.

To install from TestPyPI, run the following command:

```command
python3 -m pip install -i https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple/ fastddm
```

If one wants to test the installation multiple times with different options, please uninstall the package and clear the cache running the following commands:

```command
python3 -m pip uninstall fastddm
python3 -m pip cache remove fastddm
```

**Pull request merged**